### ISSUE TITLE
USDview on  22.11 installed on MacOS Silicon M1 Max, Ventura 13.1 has metal errors

### ISSUE THREAD

**Issue by digitalmonkey** [PixarAnimationStudios/USD#2139]
### Description of Issue
When trying to run usdview with a usd file, in a Conda environment, I get the following errors:

```
(base) ➜  dev  usdview ./USD/extras/usd/tutorials/convertingLayerFormats/Sphere.usd
Fatal coding error: Failed to create multidraw pipeline state: program_source:11:35: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  command_buffer commandBuffer [[ id(0) ]];
                                  ^~~~~
program_source:12:41: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  const device uint32_t* indexBuffer [[ id(1) ]];
                                        ^~~~~
program_source:13:44: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  const device void* mainArgumentBuffer [[ id(2) ]];
                                           ^~~~~
program_source:14:44: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  const device void* patchFactorsBuffer [[ id(3) ]];
                                           ^~~~~
program_source:15:3: error: unknown type name 'primitive_type'
  primitive_type type [[ id(4) ]];
  ^
program_source:15:26: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  primitive_type type [[ id(4) ]];
                         ^~~~~
program_source:16:36: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  StepFunctionDesc drawStepFunc [[ id(5) ]];
                                   ^~~~~
program_source:17:33: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  uint32_t numPatchStepFuncs [[ id(8) ]];
                                ^~~~~
program_source:18:3: error: no template named 'array'
  array<StepFunctionDesc, 16> patchStepFuncs [[ id(9) ]];
  ^
program_source:18:49: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  array<StepFunctionDesc, 16> patchStepFuncs [[ id(9) ]];
                                                ^~~~~
program_source:19:32: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  uint32_t numControlPoints [[ id(57) ]];
                               ^~~~~~
program_source:20:41: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  uint32_t patchBaseVertexByteOffset [[ id(58) ]];
                                        ^~~~~~
program_source:21:26: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  uint32_t numBuffers [[ id(59) ]];
                         ^~~~~~
program_source:22:3: error: no template named 'array'
  array<const device void*, 64> buffers [[ id(60) ]];
  ^
program_source:22:44: error: 'id' attribute requires Metal language standard macos-metal2.0 or higher
  array<const device void*, 64> buffers [[ id(60) ]];
                                           ^~~~~~
program_source:25:10: error: unknown type name 'render_command'
  thread render_command& cmd,
         ^
program_source:60:1: error: type 'device ArgBuffer *' is not valid for attribute 'buffer'
device ArgBuffer *args [[ buffer(26) ]])
^                         ~~~~~~~~~~
program_source:60:8: note: type 'ArgBuffer' cannot be used in buffer pointee type
device ArgBuffer *args [[ buffer(26) ]])
       ^
program_source:62:3: error: unknown type name 'render_command'
  render_command cmd(args->commandBuffer, drawItemIndex);
  ^
program_source:92:1: error: type 'device ArgBuffer *' is not valid for attribute 'buffer'
device ArgBuffer *args [[ buffer(26) ]])
^                         ~~~~~~~~~~
program_source:92:8: note: type 'ArgBuffer' cannot be used in buffer pointee type
device ArgBuffer *args [[ buffer(26) ]])
       ^
program_source:94:3: error: unknown type name 'render_command'
  render_command cmd(args->commandBuffer, drawItemIndex);
  ^
program_source:123:10: error: unknown type name 'render_command'
  thread render_command& cmd,
         ^
program_source:151:1: error: type 'device ArgBuffer *' is not valid for attribute 'buffer'
device ArgBuffer *args [[ buffer(26) ]])
^                         ~~~~~~~~~~
program_source:151:8: note: type 'ArgBuffer' cannot be used in buffer pointee type
device ArgBuffer *args [[ buffer(26) ]])
       ^
program_source:153:3: error: unknown type name 'render_command'
  render_command cmd(args->commandBuffer, drawItemIndex);
  ^
program_source:164:19: error: unknown type name 'MTLTriangleTessellationFactorsHalf'
    (const device MTLTriangleTessellationFactorsHalf*)args->patchFactorsBuffer);
                  ^
program_source:168:1: error: type 'device ArgBuffer *' is not valid for attribute 'buffer'
device ArgBuffer *args [[ buffer(26) ]])
^                         ~~~~~~~~~~
program_source:168:8: note: type 'ArgBuffer' cannot be used in buffer pointee type
device ArgBuffer *args [[ buffer(26) ]])
       ^
program_source:170:3: error: unknown type name 'render_command'
  render_command cmd(args->commandBuffer, drawItemIndex);
  ^
program_source:182:19: error: unknown type name 'MTLTriangleTessellationFactorsHalf'
    (const device MTLTriangleTessellationFactorsHalf*)args->patchFactorsBuffer);
                  ^
program_source:186:1: error: type 'device ArgBuffer *' is not valid for attribute 'buffer'
device ArgBuffer *args [[ buffer(26) ]])
^                         ~~~~~~~~~~
program_source:186:8: note: type 'ArgBuffer' cannot be used in buffer pointee type
device ArgBuffer *args [[ buffer(26) ]])
       ^
program_source:188:3: error: unknown type name 'render_command'
  render_command cmd(args->commandBuffer, drawItemIndex);
  ^
program_source:199:19: error: unknown type name 'MTLQuadTessellationFactorsHalf'
    (const device MTLQuadTessellationFactorsHalf*)args->patchFactorsBuffer);
                  ^
program_source:203:1: error: type 'device ArgBuffer *' is not valid for attribute 'buffer'
device ArgBuffer *args [[ buffer(26) ]])
^                         ~~~~~~~~~~
program_source:203:8: note: type 'ArgBuffer' cannot be used in buffer pointee type
device ArgBuffer *args [[ buffer(26) ]])
       ^
program_source:205:3: error: unknown type name 'render_command'
  render_command cmd(args->commandBuffer, drawItemIndex);
  ^
program_source:217:19: error: unknown type name 'MTLQuadTessellationFactorsHalf'
    (const device MTLQuadTessellationFactorsHalf*)args->patchFactorsBuffer);
                  ^
 [python], in _GetFunction(), /Users/digitalmonkey/dev/USD/pxr/imaging/hgiMetal/indirectCommandEncoder.mm:411
[1]    80708 abort      usdview ./USD/extras/usd/tutorials/convertingLayerFormats/Sphere.usd
```

### Steps to Reproduce
1. Build usd 22.11 with the following build script:
python build_scripts/build_usd.py /opt/local/USD --build-args USD,"-DPXR_PY_UNDEFINED_DYNAMIC_LOOKUP=ON"
the -D argument is required for Conda based python environments.
2. Attempted to run usdview:
 usdview ./USD/extras/usd/tutorials/convertingLayerFormats/Sphere.usd
3. Error messages ensue.

### System Information (OS, Hardware)
MacOS Ventura 13.1, MacBook Pro M1 Max

### Package Versions
USD 22.11
Conda python environment with Python 3.9

### Build Flags
python build_scripts/build_usd.py /opt/local/USD --build-args USD,"-DPXR_PY_UNDEFINED_DYNAMIC_LOOKUP=ON"

**Comment by dgovil** [comment]
What Qt/PySide2/PySide6 version are you using? 

I can find one example of a similar error here with an older version of Qt https://github.com/kakashidinho/metalangle/issues/36#issuecomment-744599623 but not sure if that helps. 

It might be worth trying to create a fresh conda environment and see if you get the error there. Also make sure to clear out any build directories in case they're pointing to older versions of any system libraries. E.g cmake's caches might be pointing to an older SDK location

**Comment by digitalmonkey** [comment]
I'm on pyside6 version 6.4.1, no pyside2.
I must have messed up my configuration, because I did at one point build and run usdview using the Apple system python3. However, I needed to get usd working in a condo environment, and now I consistently can't run usdview with a usda file, and get the errors related to macOS-metal2.0 above. It looks like the metal SDK I have on my MacBook is an older version. I'll look at the link above.  Thanks.

**Comment by digitalmonkey** [comment]
I was finally able to get usdview to work without the Metal errors. The additional steps I did were clean out anaconda and reinstall it,  delete my usd cloned repository and re-clone it. I also ran in my condo environment :
conda install make numpy

I confirmed that pyside6 and PyOpenGL were also installed.

Not sure which combination of these factors fixed it, but I think cleaning out my conda environments and usd folder was the likely important step.
